ci/unit/required_packages.sh

```diff
@@ -37,23 +38,6 @@ pkgs="argobots                         \
       spdk-devel                       \
       valgrind-devel"
 
-if $quick_build; then
-    if ! read -r mercury_version < "$distro"-required-mercury-rpm-version; then
-        echo "Error reading from $distro-required-mercury-rpm-version"
-        if ! ls -l "$distro"-required-mercury-rpm-version; then
-            ls -l
-        fi
-        cat "$distro"-required-mercury-rpm-version
-        exit 1
-    fi
-    pkgs+=" spdk-tools mercury\ \>=\ $mercury_version"
```

Comment:
If you don't need this mercury then we don't need the `"$distro"-required-mercury-rpm-version` file and it's creation in https://github.com/daos-stack/daos/blob/c7ad81d79458344b8fc1b69ab2b4af0da840ca1b/ci/rpm/build_success.sh#L47C1-L48 should be removed also.

Reply:
I've a feeling I tried that before and it's required by pipeline-lib which is frustrating.  Do you mind if I look at this seperately?

Reply:
I created https://github.com/daos-stack/daos/pull/13168 which is this PR with those additional two lines removed and as I thought it failed to build with errors from pipeline-lib so I think we might be stuck with this.

Reply:
On the other hand I could change that code to just create the file and add a comment as to why it's needed, I'd rather do that in a follow-up PR however.

Reply:
I was trying to make progress on this without modifying `pipeline-lib`.

Reply:
Awesome.  I'll update this PR tomorrow.

Reply:
Done.  This PR now passes without that, the `pipeline-lib` PR will need to land and then I'll have to update this again to remove the branch reference however.